stdlib/Profile/src/Allocs.jl

```diff
@@ -15,13 +15,40 @@ using Base: InterpreterIP
 # The C jl_bt_element_t object contains either an IP pointer (size_t) or a void*.
 const BTElement = Csize_t;
 
-# matches jl_raw_backtrace_t on the C side
+"""
+    RawBacktrace
+
+Represent the raw backtrace information as it is captured at the C level within Julia. 
+It holds a pointer to the raw backtrace data (`data`) and the size of this data (`size`). 
+This struct mirrors the internal C representation used by Julia to manage backtraces, 
+facilitating the transfer of data between Julia and its underlying C implementation 
+for memory allocation profiling.
+
+# Fields
+- `data::Ptr{BTElement}`: A pointer to the beginning of the backtrace data array.
+- `size::Csize_t`: The number of elements in the backtrace data array.
+"""
 struct RawBacktrace
     data::Ptr{BTElement} # in C: *jl_bt_element_t
     size::Csize_t
 end
 
-# matches jl_raw_alloc_t on the C side
+"""
+    RawAlloc
+
+Represent a single memory allocation event's raw data, as captured at the C level within Julia. 
+It includes information about the type of the allocated object, the backtrace associated with the allocation event, 
+the size of the allocation, the task in which the allocation occurred, and a timestamp marking the allocation time. 
+This struct is designed to closely mirror the corresponding C structure used internally by Julia, 
+making it easier to process allocation data in Julia's environment.
```

Comment:
Again, would be good to link functions that return a `RawAlloc` or where you get it, and what functions it can be used with?

Reply:
See also https://github.com/JuliaLang/julia/pull/53559#discussion_r1510405401